content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,430 @@
+
+### Removal of <inlinecode>jsonProtocol</inlinecode> preview feature
+
+The `jsonProtocol` preview feature is now generally available! When upgrading to Prisma 5, make sure to remove `jsonProtocol` from your preview features, if added.
+
+Prisma 4 and lower:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["jsonProtocol", "tracing"]
+}
+```
+
+Prisma 5:
+
+```prisma highlight=3;delete
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["tracing"]
+}
+```
+
+In addition, several error messages have been improved. The following error message in Prisma 4 and below
+
+```text
+Argument bool: Got invalid value 'true' on prisma.createOneD. Provided String, expected Boolean.
+Argument byteA: Got invalid value 'hello prisma ⚡️🚀' on prisma.createOneD. Provided String, expected Bytes.
+Argument xml: Got invalid value 123 on prisma.createOneD. Provided Int, expected String.
+```
+
+becomes the following in Prisma 5:
+
+```text
+Argument `bool`: Invalid value provided. Expected Boolean, provided String.
+```
+
+### <inlinecode>cockroachdb</inlinecode> provider is now required when connecting to a CockroachDB database
+
+### Removal of <inlinecode>runtime/index.js</inlinecode> from generated client
+
+#### Using public APIs from <inlinecode>runtime/index.js</inlinecode>
+
+Importing from `@prisma/client/runtime` is no longer available in Prisma 5. If you are using public APIs available in this namespace you can instead import `Prisma` and access them. For example:
+
+```js
+import { Decimal } from '@prisma/client/runtime'
+```
+
+will need to be changed to
+
+```js
+import { Prisma } from '@prisma/client'
+Prisma.Decimal
+```
+
+#### Using private APIs for a specific runtime
+
+Instead of importing private APIs from the top-level `@prisma/client/runtime` you will need to import from the runtime specific to your database engine. For example:
+
+```text
+@prisma/client/runtime/data-proxy
+```
+
+## Other changes
+
+The following changes may cause an application to initially throw an error message after upgrading to Prisma 5. Fortunately, they are easy to solve, as the underlying functionality has been removed for a while or the change is a simple string replace.
+
+### Removal of deprecated Prisma CLI flags
+
+Several deprecated CLI flags have been removed.
+
+- `--preview-feature` used in `db execute`, `db seed`, and `db diff`
+- `--experimental` and `--early-access-feature` used in `migrate`
+- `--force`/`-f` used in `db push`
+- `--experimental-reintrospection` and `--clean` used in `db pull`
+
+The use of `db push --force` can be replaced with the newer implementation `db push --accept-data-loss`.
+
+All other flags are from previous APIs and are no longer needed.
+
+### Removal of the <inlinecode>beforeExit</inlinecode> hook from the library engine
+
+Instead of the existing `beforeExit` hook we recommend using built-in Node.js exit events.
```

Comment:
Usually our sections again start with a repetition what actually cahnged, even when the headline already mentioend it.